llvm/include/llvm/TargetParser/Triple.h

```diff
@@ -831,6 +831,8 @@ class Triple {
            getArch() == Triple::spirv;
   }
 
+  bool isSPIROrSPIRV() const { return isSPIR() || isSPIRV(); }
```

Comment:
This can benefit from a comment (either a code comment or PR comment)
// Currently, we use spir-based target triples to represent JIT compilation targets for SYCL. We will transition to using spir-v based target triples to represent JIT compilation targets for SYCL very soon. This helper function is used (instead of isSPIR) to ease that transition.